### PR TITLE
Scrub PSP docs for 1.25

### PR DESCRIPTION
**Pull request by liggitt**
PodSecurityPolicy admission is dropped in 1.25 in https://github.com/kubernetes/kubernetes/pull/109798

This updates docs for 1.25 to indicate removal and truncates the PSP docs that aren't oriented at migration.

/sig auth
/cc @deads2k @tallclair

### CODE REVIEW

File: content/en/docs/concepts/security/pod-security-policy.md [hunk 1]
```diff
@@ -12,736 +12,9 @@ weight: 30
...
-This is an example of a restrictive policy that requires users to run as an
-unprivileged user, blocks possible escalations to root, and requires use of
-several security mechanisms.
-
-{{< codenew file="policy/restricted-psp.yaml" >}}
-
-See [Pod Security Standards](/docs/concepts/security/pod-security-standards/#policy-instantiation)
-for more examples.
-
-## Policy Reference
```
Comment (tallclair) [inline content/en/docs/concepts/security/pod-security-policy.md:469]: Do you think the policy reference could be useful to people looking to recreate their PSPs elsewhere?

File: content/en/docs/reference/access-authn-authz/admission-controllers.md [hunk 2]
```diff
@@ -699,24 +699,14 @@ admission plugin, which allows preventing pods from running on specifically tain
 
 {{< feature-state for_k8s_version="v1.23" state="beta" >}}
 
-This is the replacement for the deprecated [PodSecurityPolicy](#podsecuritypolicy) admission controller
-defined in the next section. This admission controller acts on creation and modification of the pod and
+This is the replacement for the deprecated `PodSecurityPolicy` admission controller.
```
Comment (tallclair) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:702]:
```suggestion
This is the replacement for the removed `PodSecurityPolicy` admission controller.
```
or "former"?

File: content/en/docs/concepts/security/pod-security-standards.md [hunk 3]
```diff
@@ -452,12 +452,6 @@ of individual policies are not defined here.
 - {{< example file="security/podsecurity-baseline.yaml" >}}Baseline namespace{{< /example >}}
 - {{< example file="security/podsecurity-restricted.yaml" >}}Restricted namespace{{< /example >}}
 
-[**PodSecurityPolicy**](/docs/concepts/security/pod-security-policy/) (Deprecated)
-
-- {{< example file="policy/privileged-psp.yaml" >}}Privileged{{< /example >}}
-- {{< example file="policy/baseline-psp.yaml" >}}Baseline{{< /example >}}
-- {{< example file="policy/restricted-psp.yaml" >}}Restricted{{< /example >}}
```
Comment (tallclair) [inline content/en/docs/concepts/security/pod-security-standards.md:459]: I think you can delete these example files. https://github.com/kubernetes/website/tree/main/content/en/examples/policy

File: content/en/docs/concepts/security/pod-security-policy.md (hunk 1 above)
Comment (liggitt) [reply content/en/docs/concepts/security/pod-security-policy.md:469]: historical content will be available at https://v1-24.docs.kubernetes.io/ … not sure we should keep info from non-current versions at HEAD

File: content/en/docs/reference/access-authn-authz/admission-controllers.md (hunk 2 above)
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:702]: I think I prefer “former”. Or:
```suggestion
Older versions of Kubernetes included an admission controller for
the beta PodSecurityPolicy API; the Pod Security admission controller
provides similar, but not identical, security enforcement.
```

Comment (sftim) [reply content/en/docs/reference/access-authn-authz/admission-controllers.md:702]: I think I prefer “former”. Or:
```suggestion
Older versions of Kubernetes included an admission controller for
the beta PodSecurityPolicy API; the Pod Security admission controller
provides similar, but not identical, security enforcement.
```

File: content/en/docs/reference/glossary/pod-security-policy.md [hunk 4]
```diff
@@ -17,4 +17,4 @@ tags:
 
 A cluster-level resource that controls security sensitive aspects of the Pod specification. The `PodSecurityPolicy` objects define a set of conditions that a Pod must run with in order to be accepted into the system, as well as defaults for the related fields. Pod Security Policy control is implemented as an optional admission controller.
 
-PodSecurityPolicy is deprecated as of Kubernetes v1.21, and will be removed in v1.25. We recommend migrating to [Pod Security Admission](/docs/concepts/security/pod-security-admission/), or a 3rd party admission plugin.
+PodSecurityPolicy was deprecated as of Kubernetes v1.21, and removed in v1.25. We recommend migrating to [Pod Security Admission](/docs/concepts/security/pod-security-admission/), or a 3rd party admission plugin.
```
[the remark this answers is not in the document]
Comment (liggitt) [reply content/en/docs/reference/glossary/pod-security-policy.md:20]: reworded

File: content/en/docs/concepts/security/pod-security-standards.md (hunk 3 above)
Comment (liggitt) [reply content/en/docs/concepts/security/pod-security-standards.md:459]: I'll leave this alone for now based on @sftim's comment

**Comment by liggitt** [comment]
addressed comments in a separate commit

**Comment by sftim** [comment]
I suggest waiting for  #34675 to merge **and** to appear on the _dev-1.25_ branch before we approve these changes - that way, there's less risk of missing a detail.

**Comment by sftim** [comment]
https://github.com/kubernetes/website/pull/34675 has merged, so the next resync to _dev-1.25_ should include that change.

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 5]
```diff
@@ -21,13 +21,12 @@ behavior of pods in a clear, consistent fashion.
 
 As a beta feature, Kubernetes offers a built-in _Pod Security_ {{< glossary_tooltip
 text="admission controller" term_id="admission-controller" >}}, the successor
-to [PodSecurityPolicies](/docs/concepts/security/pod-security-policy/). Pod security restrictions
+to the former `PodSecurityPolicy` API. Pod security restrictions
```
Comment (shannonxtreme) [inline content/en/docs/concepts/security/pod-security-admission.md:24]:
```suggestion
to the `PodSecurityPolicy` API, which was removed in Kubernetes v1.25. Pod security restrictions
```

File: content/en/docs/concepts/security/pod-security-admission.md [hunk 6]
```diff
@@ -21,13 +21,12 @@ behavior of pods in a clear, consistent fashion.
...
 are applied at the {{< glossary_tooltip text="namespace" term_id="namespace" >}} level when pods
 are created.
 
 {{< note >}}
-The PodSecurityPolicy API is deprecated and will be 
-[removed](/docs/reference/using-api/deprecation-guide/#v1-25) from Kubernetes in v1.25.
+The PodSecurityPolicy API was [removed](/docs/reference/using-api/deprecation-guide/#v1-25) from Kubernetes in v1.25.
 {{< /note >}}
```
Comment (shannonxtreme) [inline content/en/docs/concepts/security/pod-security-admission.md:30]: If you add the previous suggestion, you can remove this note

File: content/en/docs/reference/access-authn-authz/admission-controllers.md [hunk 7]
```diff
@@ -699,23 +699,16 @@ admission plugin, which allows preventing pods from running on specifically tain
...
-
-{{< feature-state for_k8s_version="v1.21" state="deprecated" >}}
-
-This admission controller acts on creation and modification of the pod and determines if it should be admitted
-based on the requested security context and the available Pod Security Policies.
-
-See also the [PodSecurityPolicy](/docs/concepts/security/pod-security-policy/) documentation
-for more information.
+Versions of Kubernetes prior to 1.25 included an admission controller for
+the beta PodSecurityPolicy API; the Pod Security admission controller
```
Comment (shannonxtreme) [inline content/en/docs/reference/access-authn-authz/admission-controllers.md:710]:
```suggestion
the beta `PodSecurityPolicy` API; the `PodSecurity` admission controller
```

File: content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md [hunk 8]
```diff
@@ -9,7 +9,7 @@ weight: 95
 
 <!-- overview -->
 The tables below enumerate the configuration parameters on
-[PodSecurityPolicy](/docs/concepts/policy/pod-security-policy/) objects, whether the field mutates
+PodSecurityPolicy objects, whether the field mutates
```
Comment (shannonxtreme) [inline content/en/docs/reference/access-authn-authz/psp-to-pod-security-standards.md:12]:
```suggestion
`PodSecurityPolicy` objects, whether the field mutates
```

**Review by shannonxtreme — APPROVED** [review]
Little nits

**Comment by cathchu** [comment]
Hey @liggitt Docs Shadow here 👋 ! There seems to be some conflicts in this branch that need to be resolved before this PR can be merged, as well as some comments that have not been resolved. Please take a look when you can so we can merge!

**Comment by liggitt** [comment]
looks like https://github.com/kubernetes/website/pull/35618 merged in the meantime... not sure how much of this PR still applies

**Comment by liggitt** [comment]
rebased and applied formatting suggestions

**Pull request merged**